compose/ui/ui-test/src/uikitMain/kotlin/androidx/compose/ui/test/ComposeUiTest.uikitMain.test.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.test
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithContent
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.uikit.ComposeUIViewControllerConfiguration
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.ComposeUIViewController
+import kotlin.test.assertEquals
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.cinterop.useContents
+import platform.CoreGraphics.CGRectMake
+import platform.Foundation.NSDate
+import platform.Foundation.NSNotification
+import platform.Foundation.NSNotificationCenter
+import platform.Foundation.NSNumber
+import platform.Foundation.NSRunLoop
+import platform.Foundation.NSValue
+import platform.Foundation.dateWithTimeIntervalSinceNow
+import platform.Foundation.runUntilDate
+import platform.Foundation.timeIntervalSinceReferenceDate
+import platform.UIKit.UIKeyboardAnimationCurveUserInfoKey
+import platform.UIKit.UIKeyboardAnimationDurationUserInfoKey
+import platform.UIKit.UIKeyboardFrameEndUserInfoKey
+import platform.UIKit.UIKeyboardWillChangeFrameNotification
+import platform.UIKit.UIKeyboardWillShowNotification
+import platform.UIKit.UIScreen
+import platform.UIKit.UIWindow
+import platform.UIKit.valueWithCGRect
+
+@ExperimentalTestApi
+fun runUIKitComposeUiTest(
+    testBlock: UIKitComposeUiTest.() -> Unit
+) {
+    UIKitComposeUiTest().runTest(testBlock)
+}
+
+@ExperimentalTestApi
+class UIKitComposeUiTest {
+    val density = Density(density = UIScreen.mainScreen().scale.toFloat())
+    val window = UIWindow(frame = UIScreen.mainScreen().bounds())
+    val screenSize = DpSize(
+        width = UIScreen.mainScreen().bounds().useContents { size.width.dp },
+        height = UIScreen.mainScreen().bounds().useContents { size.height.dp }
+    )
```

Comment:
Let's avoid multiple calls of `UIScreen.mainScreen()` function

Reply:
Done